modules/apps/object/object-api/src/main/java/com/liferay/object/entry/util/ObjectEntryThreadLocal.java

```diff
@@ -22,6 +26,19 @@ public static void addValidatedObjectEntryId(long objectEntryId) {
 		validatedObjectEntryIds.add(objectEntryId);
 	}
 
+	public static <T> T clearExpandoValues(Supplier<T> supplier) {
+		try {
+			return supplier.get();
+		}
+		finally {
+			_expandoValues.remove();
+		}
+	}
+
+	public static Map<String, Serializable> getExpandoValues() {
+		return _expandoValues.get();
+	}
```

Comment:
@JhossephG since we are only calling the `clear` after calling the `get`, maybe it's better to simplify to:

```
public static Map<String, Serializable> getExpandoValues() {
     Map<String, Serializable> expandoValues = _expandoValues.get();

     _expandoValues.remove();

     return expandoValues;
}
```

following the same pattern as [ClusterableContextThreadLocal](https://github.com/liferay-bpm/liferay-portal/blob/8a6c199560a85b1f3210a27c0fac46c9faf7d6e1/portal-kernel/src/com/liferay/portal/kernel/cluster/ClusterableContextThreadLocal.java#L20-L26)